apps/root/src/pages/transfer/components/recipient-address/index.tsx

```diff
@@ -7,22 +7,51 @@ import { useTransferState } from '@state/transfer/hooks';
 import { defineMessage, useIntl } from 'react-intl';
 import useValidateAddress from '@hooks/useValidateAddress';
 import AddressInput from '@common/components/address-input';
+import useStoredContactList from '@hooks/useStoredContactList';
 
-const RecipientAddress = () => {
+type RecipientAddressProps = Omit<ReturnType<typeof useValidateAddress>, 'address'>;
+
+const RecipientAddress = ({
+  setAddress,
+  validationResult: { errorMessage, isValidAddress },
+}: RecipientAddressProps) => {
   const intl = useIntl();
   const dispatch = useAppDispatch();
   const replaceHistory = useReplaceHistory();
+  const contactList = useStoredContactList();
   const { token, recipient: storedRecipient } = useTransferState();
   const currentNetwork = useCurrentNetwork();
-  const {
-    validationResult: { isValidAddress, errorMessage },
-    setAddress: setInputAddress,
-  } = useValidateAddress({
-    restrictActiveWallet: true,
-  });
+
+  const recognizedRecipient = React.useMemo(() => {
+    const foundContact = contactList.find((contact) => contact.address === storedRecipient);
+
+    if (!foundContact) {
+      return;
+    }
+
+    if (!foundContact.label?.label) {
+      return intl.formatMessage(
+        defineMessage({
+          defaultMessage: 'This address matches a contact of yours',
```

Comment:
Wait if we dont have a label is it really a contact? Or does this refer to one of the addresses you have added but dont have a label?

Reply:
Second one 😄 . It could be a contact/linkedWallet without label. It seemed strange to show 'this address matches with 0x...'

Reply:
yeah lets leave the text change for @0xged to handle, I would go with something like `This address matches one of your added wallets`

Reply:
There is still a case where you can add a contact without a label